### PR TITLE
[App Service] `az webapp up`: Show status during deployment for linux apps

### PR DESCRIPTION
**Pull request by StrawnSC**
**Related command**

- `az webapp up`
- `az webapp deployment source config-zip`

**Description**

Uses the new Web App deployment status API to give the user updates on the their status of their deployments 

**Testing Guide**

**History Notes**

[App Service] `az webapp up`: Show status during deployment for linux apps
[App Service] `az webapp deployment source config-zip`: Show status during deployment for linux apps

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/azure/cli/command_modules/appservice/utils.py [hunk 1]
```diff
@@ -266,3 +267,24 @@ def is_webapp(app):
...
+        sys.stdout.write(self.prefix + self.tickers[self.currTicker] + self.suffix)
+        sys.stdout.flush()
+        self.currTicker += 1
+        self.currTicker = self.currTicker % len(self.tickers)
+
+    @classmethod
+    def flush(cls):
+        sys.stdout.flush()
+        sys.stdout.write('\r')
+        sys.stdout.write("\033[K")
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/appservice/utils.py:290]: Better to add a annotation here.
eg., \33[K: clears from cursor to end of line.

**Review by zhoxing-ms — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Comment by DavidZidar** [comment]
@StrawnSC Hi, this feature is causing some issues, see #23925. There needs to be an option way to skip this waiting.

**Comment by panchagnula** [comment]
> @StrawnSC Hi, this feature is causing some issues, see #23925. There needs to be an optional way to skip this waiting.

@DavidZidar thanks for the feedback - the waiting message is from the API, checking the deployment status, & depends on how long the deployment is taking. I see the issue as well linked here.

**Comment by DavidZidar** [comment]
@panchagnula In my case the waiting message keeps printing for several minutes after the site is already running. I don't know where the issue is but something is not right and it was working fine before this release.

**Comment by panchagnula** [comment]
> @panchagnula In my case the waiting message keeps printing for several minutes after the site is already running. I don't know where the issue is but something is not right and it was working fine before this release.

Got it. Thank you. We will reach out to the owners of this API on our end as well, & we considering the best options here  for CLI command, in the meanwhile. Thanks for bringing this to our attention

**Pull request merged**